src/client/QXmppSendStanzaParams.cpp

```diff
@@ -52,3 +53,27 @@ void QXmppSendStanzaParams::setEncryptionJids(QVector<QString> encryptionJids)
 {
     d->encryptionJids = std::move(encryptionJids);
 }
+
+///
+/// Returns the possible trust levels a key must have to be used for encryption.
+///
+/// If no trust levels are set, the encryption manager uses an own default.
+///
+/// \return the trust levels of the keys used for encryption
+///
+QXmppTrustStorage::TrustLevels QXmppSendStanzaParams::acceptedTrustLevels() const
+{
+    return d->acceptedTrustLevels;
+}
+
+///
+/// Sets the possible trust levels a key must have to be used for encryption.
+///
+/// If no trust levels are set, the encryption manager uses an own default.
+///
+/// \param trustLevels trust levels of the keys used for encryption
+///
+void QXmppSendStanzaParams::setAcceptedTrustLevels(QXmppTrustStorage::TrustLevels trustLevels)
+{
+    d->acceptedTrustLevels = trustLevels;
+}
```

Comment:
```suggestion

///
/// Returns the possible trust levels a key must have to be used for encryption.
///
/// If no trust levels are set, the encryption manager uses an own default.
///
/// \return the trust levels of the keys used for encryption
///
std::optional<QXmppTrustStorage::TrustLevels> QXmppSendStanzaParams::acceptedTrustLevels() const
{
    if (d->acceptedTrustLevels) {
        return d->acceptedTrustLevels;
    }
    return {};
}

///
/// Sets the possible trust levels a key must have to be used for encryption.
///
/// If no trust levels are set, the encryption manager uses an own default.
///
/// \param trustLevels trust levels of the keys used for encryption
///
void QXmppSendStanzaParams::setAcceptedTrustLevels(std::optional<QXmppTrustStorage::TrustLevels> trustLevels)
{
    d->acceptedTrustLevels = trustLevels.value_or(0);
}
```

Allows to check whether trust levels have been set or the default should be used.